pre_commit_hooks/check_for_go_fips_test.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import argparse
+import sys
+from collections.abc import Sequence
+from typing import Any
+
+import ruamel.yaml
+
+yaml = ruamel.yaml.YAML(typ="safe")
+
+
+def check_go_fips_compliance(melange_cfg: dict[str, Any]) -> tuple[bool, list[str]]:
+    """
+    Check if all go-fips usages have corresponding tests.
+    Returns (is_compliant, list_of_missing_tests).
+    """
+    issues = []
+
+    # Check if main package uses go-fips
+    main_uses_fips = False
+    main_has_test = False
+
```

Comment:
I feel like we shouldn't need to duplicate the code between main package parsing and subpkg parsing. See the shellcheck test, where it puts the main package and any sub package in a list, and just iterates through them all the same way. We should probably consider creating a library class that just does stuff like this for us so we can share it.